robot/impl/local_robot.go

```diff
@@ -1251,7 +1259,22 @@ func (r *localRobot) restartSingleModule(ctx context.Context, mod config.Module)
 		Left:     r.Config(),
 		Right:    r.Config(),
 		Added:    &config.Config{},
-		Modified: &config.ModifiedConfigDiff{Modules: []config.Module{mod}},
+		Modified: &config.ModifiedConfigDiff{},
+		Removed:  &config.Config{Modules: []config.Module{mod}},
+	}
+	err := r.manager.updateResources(ctx, &diff)
+	if err != nil {
+		return err
+	}
+	err = r.localPackages.RecopyIfChanged(ctx, mod)
+	if err != nil {
+		return err
+	}
```

Comment:
[🏎️?]  I wonder what happens if config-polling happens during this Recopy and updateResources is run....


I think you should move this into `updateResources` if possible....

But also happy to wait until someone hits this

Reply:
thinking through this. we are doing `[stop, replace, start]` with no locking in the middle. bad concurrency cases are:
- we overwrite files while they're being executed. in many cases this will be fine, but I suspect not all. in cases where there's a lock, I suspect the Recopy() call will fail and the CLI restart command will return an error, which may or may not leave things in a bad state
- if you try to start a module that is already started, you get 'not adding module that already exists' [modmanager/manager.go#L255](https://github.com/viamrobotics/rdk/blob/a1a84d8220da6e7a28590397b254a83c397e1234/module/modmanager/manager.go#L255)
- if it's not a local tarball module, i.e. you're trying to restart a normal module, it may fail from double-start, 

For pure stop-start overlaps (i.e. for non-tarball modules where no 'install' step happens), I think the unique cases are:
- :stop_sign: :checkered_flag: :stop_sign: :checkered_flag: 
- :stop_sign: :stop_sign: :checkered_flag: :checkered_flag: 

:point_up: both of those leave the module started, with a possible spurious error from double-start.

I don't think any of this will permanently break other modules (i.e. other than the module you are restarting). This path shouldn't affect production robots that aren't using local tarballs.

My vote: ship as-is to unblock other work, then look into module state + restart-versioning so we can use updateResources normally

Open to other ideas